### PR TITLE
feat: Introduce - finish state machine

### LINKED ISSUE #883: Introduce - possibility to finish state machine
Need to add a method to the introduce service `InvitationReceived` which should be executed by the did-exchange service or client to finish the state machine

### PR DESCRIPTION
**Pull request by soluchok**
* added `InvitationReceived` function to  finish state machine
* the behavior of the state machine is completed

Closes #883 
Part of #269

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/didcomm/protocol/introduce/service.go [hunk 1]
```diff
@@ -108,7 +112,7 @@ type Provider interface {
 // Forwarder provides the possibility to forward an invitation
 type Forwarder interface {
 	// method should be implemented in didexchange service
-	SendInvitation(inv *didexchange.Invitation, dest *service.Destination) error
+	SendInvitation(pthID string, inv *didexchange.Invitation, dest *service.Destination) error
```
Comment (troyronda) [inline pkg/didcomm/protocol/introduce/service.go]: Are you planning to create a new Invitation object based on `inv` and `pthID`?

File: pkg/didcomm/protocol/introduce/service.go [hunk 2]
```diff
@@ -253,6 +257,28 @@ func (s *Service) doHandle(msg *service.DIDCommMsg, outbound bool) (*metaData, e
 	}, nil
 }
 
+// InvitationReceived is used to finish the state machine
+// the function should be called by didexchange after receiving an invitation
```
Comment (troyronda) [inline pkg/didcomm/protocol/introduce/service.go]: How does DID Exchange know to do this?

File: pkg/didcomm/protocol/introduce/service.go (hunk 1 above)
Comment (soluchok) [reply pkg/didcomm/protocol/introduce/service.go]: To close the state machine I need the same thread id. According to that, I assume that did-exchange will start protocol (did-exchange) with parent thread id.

File: pkg/didcomm/protocol/introduce/service.go (hunk 2 above)
Comment (soluchok) [reply pkg/didcomm/protocol/introduce/service.go]: I don't know yet. Probably after receiving an invitation did-exchange should check parent thread id. Somehow it should determine that pthid was provided by the introduce protocol.

Comment (troyronda) [reply pkg/didcomm/protocol/introduce/service.go]: We will need to avoid a direct coupling.

**Pull request merged**